src/mjolnir/shortcutbuilder.cc

```diff
@@ -218,6 +218,12 @@ bool CanContract(GraphReader& reader,
   if (!EdgesMatch(tile, edge1, edge2))
     return false;
 
+  //if edges are greater than the maximum legnth constnat, do not contract them
+  uint32_t totalLength = edge1->length() + edge2->length(); 
+  if(totalLength > kMaxEdgeLength) {
+    return false; 
+  }
+
```

Comment:
is this one necessary? is there a case in the main function where it might become a problem if we don't add this check? otherwise I'd just throw it